### DISCUSSION TITLE
Adding ssh key to the creation process

### DISCUSSION THREAD

**Discussion by Herredm1** [proxmoxer/proxmoxer#157] [Q&A]
Below is an example of a config file for a container-creating script. The config below works, except for the ssh option. Is it possible to add the ssh key, if so, how?   Thanks in advance!

proxmox.nodes(node_name).lxc.create(
        hostname=CTName,
        vmid=ctid,
        ostemplate=template_name,
        memory=512, 
        cores=cores,  
        swap=512,  
        onboot=1, 
        password=root_password,
        nameserver = IP,
        rootfs='local-lvm:8',  
        ssh_public_keys=pubkey, <---------
        
        **network_settings    
    )

**Comment by morph027** [comment]
Looks like https://github.com/proxmoxer/proxmoxer/issues/153

**Comment by jhollowe** [comment]
You will want to look at the [Cloud-init docs](https://proxmoxer.github.io/docs/2.0/examples/cloud-init/#example-code-setting-cloud-init-options) and look at the example code for how to encode/escape the SSH key to pass through correctly to PVE
